aws-cognito-idp/src/main/java/com/workshop/awscognitoidp/controllers/ScoreController.java

```diff
@@ -0,0 +1,71 @@
+package com.workshop.awscognitoidp.controllers;
+
+import com.workshop.awscognitoidp.models.Subject;
+import com.workshop.awscognitoidp.repositories.SubjectRepository;
+import com.workshop.awscognitoidp.services.JwtDataRetriever;
+import com.workshop.awscognitoidp.services.JwtValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/api/students")
+public class ScoreController {
+    @Autowired
+    private JwtValidator jwtValidator;
+    @Autowired
+    private JwtDataRetriever jwtDataRetriever;
+    @Autowired
+    private SubjectRepository subjectRepository;
+
+    @GetMapping("/scores")
+    public List<Subject> getScores(@RequestHeader("Authorization") String token) {
+        if(jwtValidator.validateJwtToken(token.substring(7))){
+            System.out.println("TOKEN VALID");
+            String role  = jwtDataRetriever.getTokenRole(token.substring(7));
+            if (role.equals("trainer")) {
+                return subjectRepository.getAll();
+            } else {
+                throw new AccessDeniedException("Insufficient permissions");
```

Comment:
I think... you do not need to do this control here... there are some annotations that can help you... Please refer to this [link](https://openclassrooms.com/en/courses/5683681-secure-your-web-application-with-spring-security/6695821-configure-role-based-access-control)